src/main/java/com/microsoft/aad/msal4j/Authority.java

```diff
@@ -51,15 +51,17 @@ private void setCommonAuthorityProperties() {
         this.host = canonicalAuthorityUrl.getAuthority().toLowerCase();
     }
 
-    static Authority createAuthority(URL authorityUrl){
-       AuthorityType authorityType = detectAuthorityType(authorityUrl);
-       if(authorityType == AuthorityType.AAD){
-           return new AADAuthority(authorityUrl);
-       } else if(authorityType == AuthorityType.B2C) {
-           return new B2CAuthority(authorityUrl);
-       } else {
-           throw new IllegalArgumentException("Unsupported Authority Type");
-       }
+    static Authority createAuthority(URL authorityUrl) {
+        validateAuthority(authorityUrl);
+
+        AuthorityType authorityType = detectAuthorityType(authorityUrl);
```

Comment:
Looks like `detectAuthorityType` can return AuthorityType.ADFS, but that would throw "Unsupported Authority Type" error, which is actually supported. Right?